src/orion/benchmark/task/task_wrapper.py

```diff
@@ -0,0 +1,164 @@
+from abc import ABC
+from dataclasses import asdict, is_dataclass
+from typing import (
+    Any,
+    Dict,
+    Generic,
+    List,
+    TypeVar,
+    Union,
+)
+import copy
+
+import numpy as np
+from orion.algo.space import Dimension, Space
+from orion.benchmark.task.base import BaseTask, BenchmarkTask
+from logging import getLogger as get_logger
+
+from orion.core.utils.points import flatten_dims, regroup_dims
+from orion.core.utils.format_trials import dict_to_trial
+
+from orion.core.io.space_builder import SpaceBuilder
+from orion.core.utils.format_trials import trial_to_tuple, tuple_to_trial
+from orion.core.worker.trial import Trial
+
+logger = get_logger(__name__)
+
+
+TaskType = TypeVar("TaskType", bound=BaseTask)
+InputType = TypeVar("InputType")
+
+
+class TaskWrapper(BaseTask, Generic[TaskType], ABC):
+    """ ABC for a Wrapper around a Task.
+
+    Wrappers could be used to modify the search space, trials, or the results of the task.
+    """
+
+    def __init__(self, task: TaskType, max_trials: int = None):
+        # TODO: Should Wrappers pass the `task` as an argument to the super() constructor?
```

Comment:
This is good for now. It will be adjusted anyway for release v0.2.0 as we merge #135.